src/main/java/duke/task/TaskManager.java

```diff
@@ -0,0 +1,145 @@
+package duke.task;
+
+import java.util.ArrayList;
+import duke.DukeException;
+import java.io.BufferedReader;
+import java.io.IOException;
+
+public class TaskManager {
+    // Constants
+    private static final int MAX_TASKS = 100;
+    private static final String DELIMITER = " | ";
+
+    // Task list
+    private final ArrayList<Task> tasks = new ArrayList<>();
+
+    // Class variable for counting number of tasks
+    private static int taskCount = 0;
+
+    public int getTasksCount() {
+        return taskCount;
+    }
+
+    // Add new task to the task list
+    public Task addTask(Task task) {
+        tasks.add(task);
+        taskCount++;
+        return task;
+    }
+
+    public Task deleteTask(int id) {
+        Task task = tasks.get(id);
+        tasks.remove(id);
+        taskCount--;
+        return task;
+    }
+
+    public Task addTodo(String description) {
+        Todo todo = new Todo(description);
+        return this.addTask(todo);
+    }
+
+    public Task addEvent(String description) {
+        int atIndex = description.indexOf("/at");
+        String at;
+        try {
+            at = description.substring(atIndex + 4);
+            description = description.substring(0, atIndex - 1);
+        } catch (Exception e) {
+            System.out.println("Invalid command format!!" + System.lineSeparator() +
+                    "Correct format is: " + "event <event name> /at <time>");
+            return null;
+        }
```

Comment:
Maybe parsing of information can be abstracted to a different method to enhance quality and readability.